src/Advection/vector_invariant_advection.jl

```diff
@@ -213,6 +213,12 @@ end
     return + upwind_biased_product(û, ζᴸ, ζᴿ)
 end
 
+## Upwinding `δ` is not like upwinding `ζ`. `u` is trasporting `ζ`, while 
+## `u` is the transported quantity for the `δ` term that derives from the vertical advection of `u`
+## For this reason, the divergence (`δ`) must be multiplied by the area in z to account for the flux of `w` 
+## (as done for the other advecting velocities). 
```

Comment:
It looks like you are applying the WENO reconstruction to `A * d` where `A` is cell area and `d` is divergence. Is that correct? Should we say that? (It's not obvious to me from the comment that this is the new algorithm.)

Previously, we were reconstructing `d`, not `A * d`?

Reply:
Exactly. I changed the implementation to mirror the MITgcm vector invariant form, where the metrics are included in the reconstruction of the advective velocity:
Here:
https://github.com/CliMA/Oceananigans.jl/blob/98bcdde7ae679583ea88474ec0909469abccfc90/src/Advection/vector_invariant_advection.jl#L159

and here:
https://github.com/CliMA/Oceananigans.jl/blob/98bcdde7ae679583ea88474ec0909469abccfc90/src/Advection/vector_invariant_advection.jl#L178-L179

The vorticity follows the same rule, but there was no metric term in the divergence flux

Reply:
That vorticity reconstruction scheme has something to do with discrete enstrophy or energy conservation right?

Reply:
That's what the docs claim:

https://mitgcm.readthedocs.io/en/latest/algorithm/algorithm.html#id8

Reply:
As for vertical advection there is a comment about "spurious generation of kinetic energy"

https://mitgcm.readthedocs.io/en/latest/algorithm/algorithm.html#shear-terms

Perhaps similar logic applies to divergence but it's certainly not obvious without doing the math?

Reply:
I don't think it's quite as simple as saying "the metrics are included" (or not). These are specific formulations constructed to have specific discrete properties...

Could it help to try to derive a discrete energy or potential vorticity equation?